integration_test.go

```diff
@@ -0,0 +1,284 @@
+	testProjectID = os.Getenv("HARBOURBRIDGE_TESTS_GCLOUD_PROJECT_ID")
+	testInstanceID = os.Getenv("HARBOURBRIDGE_TESTS_GCLOUD_INSTANCE_ID")
+
+	ctx := context.Background()
+	flag.Parse() // Needed for testing.Short().
+	noop := func() {}
+
+	if testing.Short() {
+		log.Println("Integration tests skipped in -short mode.")
+		return noop
+	}
+
+	if testProjectID == "" {
+		log.Println("Integration tests skipped: HARBOURBRIDGE_TESTS_GCLOUD_PROJECT_ID is missing")
+		return noop
+	}
+
+	if testInstanceID == "" {
+		log.Println("Integration tests skipped: HARBOURBRIDGE_TESTS_GCLOUD_INSTANCE_ID is missing")
+		return noop
+	}
+
+	var err error
+	databaseAdmin, err = database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		log.Fatalf("cannot create databaseAdmin client: %v", err)
+	}
+
+	return func() {
+		databaseAdmin.Close()
+	}
+}
+
+func dropDatabase(t *testing.T, dbPath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	// Drop the testing database.
+	if err := databaseAdmin.DropDatabase(ctx, &databasepb.DropDatabaseRequest{Database: dbPath}); err != nil {
+		t.Fatalf("failed to drop testing database %v: %v", dbPath, err)
+	}
+}
+
+func cleanupFiles(t *testing.T, files []string) {
+	for _, file := range files {
+		if _, err := os.Stat(file); err == nil {
+			err = os.Remove(file)
+			if err != nil {
+				t.Errorf("failed to delete file: %v", file)
+			}
+		}
+	}
+}
+
+func prepareIntegrationTest(t *testing.T) {
+	if databaseAdmin == nil {
+		t.Skip("Integration tests skipped")
+	}
+}
+
+func TestIntegration_SimpleUse(t *testing.T) {
+	t.Parallel()
+
+	prepareIntegrationTest(t)
+
+	now := time.Now()
+	dbName, _ := getDatabaseName(now)
+	dbPath := fmt.Sprintf("projects/%s/instances/%s/databases/%s", testProjectID, testInstanceID, dbName)
+
+	dataFilepath := "test_data/pg_dump.test.out"
+	filePrefix = dbName + "."
+	f, err := os.Open(dataFilepath)
+	if err != nil {
+		t.Fatalf("failed to open the test data file: %v", err)
+	}
+	err = process(testProjectID, testInstanceID, dbName, &ioStreams{f, os.Stdout}, filePrefix, now)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Drop the database later.
+	defer dropDatabase(t, dbPath)
+	defer cleanupFiles(t, []string{filePrefix + schemaFile, filePrefix + reportFile, filePrefix + badDataFile})
+
+	checkResults(t, dbPath)
+}
+
+func TestIntegration_Command(t *testing.T) {
+	t.Parallel()
+
+	prepareIntegrationTest(t)
+
+	now := time.Now()
+	dbName, _ := getDatabaseName(now)
+	dbPath := fmt.Sprintf("projects/%s/instances/%s/databases/%s", testProjectID, testInstanceID, dbName)
+
+	dataFilepath := "test_data/pg_dump.test.out"
+	filePrefix = dbName + "."
+	// Be aware that when testing with the command, the time `now` might be
+	// different between file prefixes and the contents in the files. This
+	// is because file prefixes use `now` from here (the test function) and
+	// the generated time in the files uses a `now` inside the command, which
+	// can be different.
```

Comment:
So the test will still pass (since we don't check the contents of the files), but the cleanup may fail?

Reply:
The cleanup will not fail because we explicitly pass the file prefix as a command parameter here. We can cleanup files according to the given file prefix. 

In my current version, I created a temporary directory so I can delete the directory easily. However, the main purpose of temporary directory is to solve a possible conflict in parallel tests. 

If we want to write some tests to check contents based on the time `now`, we should be aware of this issue that we can't pass a time via command line.